Firestore/core/test/unit/model/value_util_test.cc

```diff
@@ -132,6 +133,15 @@ class ValueUtilTest : public ::testing::Test {
     EXPECT_EQ(expected_canonical_id, actual_canonical_id);
   }
 
+  void VerifyDeepClone(const google_firestore_v1_Value& value) {
+    // `Message` takes ownership of the proto value and destroys it. We use
+    // it to verify that all values own their own memory.
+    nanopb::Message<google_firestore_v1_Value> clone1{DeepClone(value)};
+    nanopb::Message<google_firestore_v1_Value> clone2{DeepClone(*clone1)};
+    EXPECT_TRUE(value == *clone1);
```

Comment:
Question: when these tests fail, what do the failure messages look like? Is it clear which invocation of `VerifyDeepClone` failed?

Reply:
No, it requires debugging. I don't know how to solve this though (other than by passing in the index of the callsite). I wish there was a proper stracktrace. 

I added a better error message though, which should help a bit.

Reply:
Can you check again after replacing `EXPECT_TRUE` with `EXPECT_EQ`? One possible approach is to add [`SCOPED_TRACE`](https://github.com/google/googletest/blob/master/docs/advanced.md#using-assertions-in-sub-routines) before every invocation of this helper, but please check with `EXPECT_EQ` first to see if it's necessary. Another approach, should `EXPECT_EQ` not work out, is to log the stringified form of both values being compared upon failure.